### ISSUE TITLE
Octolock shouldn't bypass clear body.

### ISSUE THREAD

**Issue by TZubiri** [smogon/pokemon-showdown#6039]
The move octolock reduces the defenses of the opponent each turn, despite abilities like clear body (and the illegal flower veil)  preventing stat drops, octolock goes through them. It shouldn't.

Minimum verifiable example: https://replay.pokemonshowdown.com/gen8ou-1013532533

Original user report: https://www.smogon.com/forums/threads/bug-reports-v3-read-original-post-before-posting.3634749/post-8289028

**Comment by ykabusalah** [comment]
Can I tackle this issue as well?

**Comment by TZubiri** [comment]
> Can I tackle this issue as well?

Thanks for your interest in contributing!

I have assigned this issue to you. You can go ahead and start working on it, if you get stuck or have any technical questions, feel free to post them on this thread.
If you manage to fix the bug and are confident in your solution, please read [this](https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md) before requesting a pull.

Good luck!

**Comment by PartMan7** [comment]
I think it's already been done with all the other Octolock interactions.

https://github.com/smogon/pokemon-showdown/pull/6065

**Comment by TZubiri** [comment]
Thanks Partman. I'll follow that PR and close the issue once I can test it in prod.

@ykabusalah Were you able to make any progress? The issue is going to be closed once the pull request from the other dev is merged, but don't despair! You should still post your solution, later you can compare it with the implemented solution to gain insight and perhaps feedback from other devs. That way you can take on another issue in the future and keep improving your velocity.

**Comment by Zarel** [comment]
Fixed by #6065
